core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -403,6 +406,9 @@ private[spark] class TaskSchedulerImpl(
       if (!executorIdToRunningTaskIds.contains(o.executorId)) {
         hostToExecutors(o.host) += o.executorId
         executorAdded(o.executorId, o.host)
+        // Assumes the first offer will include all cores (free cores == all cores)
```

Comment:
can anyone confirm whether it is true that the first resource offer for an executor will include all cores?
even if this is true it feels odd to rely on it.